From b4b9d129bcb44f11a7d36f3a43a48d6f46b3e1fa Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Fri, 2 Dec 2016 13:34:32 -0800
Subject: [PATCH] joyent/sdc-docker#102 need to be using bootstrap mode with
 cueball / moray-client v2 Reviewed by: Trent Mick <trent.mick@joyent.com>

---
 lib/docker.js                  | 5 ++++-
 sapi_manifests/docker/template | 3 +++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/lib/docker.js b/lib/docker.js
index 8629fe3..d1d05f7 100644
--- a/lib/docker.js
+++ b/lib/docker.js
@@ -367,7 +367,10 @@ App.prototype.createMorayClient = function createMorayClient() {
         host: self.config.moray.host,
         noCache: true,
         port: self.config.moray.port,
-        reconnect: true
+        reconnect: true,
+        dns: {
+            resolvers: [self.config.binder.domain]
+        }
     };
 
     self.log.debug(conf, 'Creating moray client');
diff --git a/sapi_manifests/docker/template b/sapi_manifests/docker/template
index a056637..1bfa4b0 100644
--- a/sapi_manifests/docker/template
+++ b/sapi_manifests/docker/template
@@ -31,6 +31,9 @@
         "host": "{{{MORAY_SERVICE}}}",
         "port": 2020
     },
+    "binder": {
+        "domain": "{{{BINDER_SERVICE}}}"
+    },
     "fwapi": {
         "url": "http://fwapi.{{{datacenter_name}}}.{{{dns_domain}}}"
     },
-- 
2.21.0

